agent/api/task.go

```diff
@@ -899,6 +900,25 @@ func (task *Task) updateContainerDesiredStatus() {
 	}
 }
 
+// updateResourceDesiredStatus sets all resources' desired status's depending on the
+// task's desired status
+// TODO: Create a mapping of resource status to the corresponding task status and use it here
+func (task *Task) updateResourceDesiredStatus() {
+	for _, r := range task.Resources {
+		taskDesiredStatus := task.GetDesiredStatus()
```

Comment:
The task's desired status can change anytime. That's the reason why we check it every time here.

Reply:
Right, because of that, you have inconsistent transitions potentially being applied for different resources. It would be lot more straight forward if during one invocation of `updateResourceDesiredStatus()`, you applied one transition. Because, subsequent transitions should be caught in subsequent invocations of `updateResourceDesiredStatus`